src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -220,19 +213,24 @@ public JWTHeaderWithCertificate(IClientAssertionCertificate credential)
                 : base(credential)
             {
                 _thumbPrint = this.Credential.Thumbprint;
-                X509CertificatePublicCertValue = string.Empty;
+                X509CertificatePublicCertValue = null;
 
                 //Check to see if credential is our implementation or developer provided.
-                if (!credential.GetType().ToString().Contains("Microsoft.IdentityModel"))
+                if (credential.GetType().ToString() != "Microsoft.IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate")
                 {
                     CallState.Default.Logger.Warning(null, "The implementation of IClientAssertionCertificate is developer provided and it should be replaced with library provided implmentation.");
                     return;
                 }
 
-#if (NET45 || NETSTANDARD1_3)
+#if NET45
+                if (credential is ClientAssertionCertificate cert)
+                {
+                    X509CertificatePublicCertValue = Convert.ToBase64String(cert.Certificate.GetRawCertData());
+                }
+#elif NETSTANDARD1_3
                 if (credential is ClientAssertionCertificate cert)
                 {
-                    X509CertificatePublicCertValue = Base64UrlEncoder.Encode(cert.Certificate.ToString());
+                    X509CertificatePublicCertValue = Convert.ToBase64String(Encoding.UTF8.GetBytes(cert.Certificate.ToString()));
```

Comment:
GetRawCertData() is not available for net standard 1.3, however, the ToString() method for the certificate does return the contents of the certificate because it overwritten to do so. 

        //
        // Summary:
        //     Displays an X.509 certificate in text format.
        //
        // Returns:
        //     The certificate information.
        public override string ToString();